### ISSUE TITLE
Clear canvas in vispy.scene

### ISSUE THREAD

**Issue by iberinsk** [vispy/vispy#1288]
Hi! I am trying to do an animation in vispy.scene to visualize motion of points from existing numpy array. Now I see the lines and points at every step. How can I clear a canvas to see only current positions? 

```
import numpy as np
from vispy import app, scene

#Let us construct some test array of coordinates in time
#Number of points
N = 2
#Time of modelling
T = 100

#Create positions 
r = np.empty(2,)
for n in np.arange(N):
    r = np.vstack((r,np.random.normal([1,1])))
r = np.delete(r,0,0) # spare element 

coords = np.array(r)

for t in np.arange(T):
    r += np.random.normal(1,1)*np.sin(t)
    coords = np.concatenate((coords,r), axis = 1)

#Now let us visualize array 'coords'
canvas = scene.SceneCanvas(keys='interactive')
view = canvas.central_widget.add_view()
color = np.array([1.,1.,1.,1.]) #White
#3D axis
axis = scene.visuals.XYZAxis(parent=view.scene)


t=0
def update(ev):
    global t, pos, color, line
    t+=1
    pos = []
    lines = []
    for n in range(N):
        x = coords[n,2*t]
        y = coords[n,2*t+1]
        pos.append([x,y])
    pos = np.array(pos)
    scatter = scene.visuals.Markers()
    scatter.set_data(pos, edge_color=(1,1,1,1), face_color=(0, 0, 1, 1), size=15)
    view.add(scatter) 
    view.camera = 'turntable'  # 'arcball' or 'turntable'
    line = scene.visuals.Line(pos=pos, color=color, width = 3, parent = view.scene)
    lines.append(line)

timer = app.Timer()
timer.connect(update)
timer.start(0.05,T) #interval, iterations

canvas.show()
app.run()

```

**Comment by scottpaine** [comment]
You're adding a scatter plot every single time, rather than resetting the data. Using `scatter.set_data`, you can overwrite the old positions. The same thing exists for the `LineVisual`, too. 

So, don't create a new object every time. Just keep track of the current one and update it.

The following works fine for me:

``` python
import numpy as np
from vispy import app, scene

# Let us construct some test array of coordinates in time
# Number of points
N = 2
# Time of modelling
T = 100

# Create positions
r = np.empty(2,)
for n in np.arange(N):
    r = np.vstack((r, np.random.normal([1, 1])))
r = np.delete(r, 0, 0)  # spare element

coords = np.array(r)

for t in np.arange(T):
    r += np.random.normal(1, 1)*np.sin(t)
    coords = np.concatenate((coords, r), axis=1)

# Now let us visualize array 'coords'
canvas = scene.SceneCanvas(keys='interactive')
view = canvas.central_widget.add_view()
color = np.array([1., 1., 1., 1.])  # White
# 3D axis
axis = scene.visuals.XYZAxis(parent=view.scene)
# Scatter plot and line

# First, initialize x and y
init_x = coords[:, 0]
init_y = coords[:, 1]
init_pos = np.array([init_x, init_y])

# Now, make the markers and the line.
# Use the parent argument so they are automatically added to the scene.
scatter = scene.visuals.Markers(pos=init_pos,
                                edge_color=(1, 1, 1, 1),
                                face_color=(0, 0, 1, 1),
                                size=15,
                                parent=view.scene)
line = scene.visuals.Line(pos=init_pos,
                          color=color,
                          width=3,
                          parent=view.scene)

# Set the camera
view.camera = 'turntable'  # 'arcball' or 'turntable'

# Start the time
t = 0


def update(ev):
    global t, pos, color, line
    t += 1
    pos = []
    for n in range(N):
        x = coords[n, 2*t]
        y = coords[n, 2*t + 1]
        pos.append([x, y])
    pos = np.array(pos)
    scatter.set_data(pos,
                     edge_color=(1, 1, 1, 1),
                     face_color=(0, 0, 1, 1),
                     size=15)

    line.set_data(pos=pos, color=color, width=3)

timer = app.Timer()
timer.connect(update)
timer.start(0.05, T)  # interval, iterations

canvas.show()
app.run()
```

**Comment by iberinsk** [comment]
Thanks! But
```
scatter = scene.visuals.Markers(pos=init_pos,
                                edge_color=(1, 1, 1, 1),
                                face_color=(0, 0, 1, 1),
                                size=15,
                                parent=view.scene)
```
gives me an error 

> TypeError: __init__() got an unexpected keyword argument 'edge_color'

So, I used the following for initialization:
```
scatter = scene.visuals.Markers()
scatter.set_data(init_pos,
                     edge_color=(1, 1, 1, 1),
                     face_color=(0, 0, 1, 1),
                     size=15)
scatter.add_parent(view.scene) 
```
Also, I managed to see the merkers only after  `view.add(scatter) ` was used. Now it works for me. 

How can I authomatically update size of the canvas to see the line at each step?

**Comment by scottpaine** [comment]
Check your version of vispy. If you installed via Anaconda, you're running an old version, and there's a lot of missing parts.

Follow the installation guide under "Installation" on the [main repository page](https://github.com/vispy/vispy)

**Comment by iberinsk** [comment]
Now I use WinPython-64bit-3.4.4.4-Qt5 + vispy 0.4.0

**Comment by scottpaine** [comment]
Right, so the current branch of vispy is at 0.5.0, and has a TON of bugfixes in it and more accessibility, so perform an installation according to the instructions under "Installation" on the [main repository page](https://github.com/vispy/vispy)

**Comment by hadim** [comment]
@iberinsk waiting the 0.5.0 release you can install the version in the repo with `pip install https://github.com/vispy/vispy/archive/master.zip`

**Comment by iberinsk** [comment]
@hadim, Thank you! I have installed vispy 0.5.0 and the code of @scottpaine works perfectly on WinPython. However, I try the same code on Ubuntu using Anaconda with vispy and have the following: 

> WARNING: QXcbIntegration: Cannot create platform OpenGL context, neither GLX nor EGL are enabled
and then
> raise RuntimeError('context could not be created')
What can be wrong?

**Comment by kmuehlbauer** [comment]
> I have installed vispy 0.5.0 and the code of @scottpaine works perfectly on WinPython.

@iberinsk Closing this as resolved. Did you get your other problem solved? Please open a new issue, if this persists.
